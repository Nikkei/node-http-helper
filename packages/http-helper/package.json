{
    "name": "@nikkei/http-helper",
    "version": "2.4.1",
    "type": "module",
    "files": [
        "__dist/",
        "LICENSE",
        "package.json",
        "README.md"
    ],
    "typings": "./__dist/mod.d.ts",
    "exports": {
        ".": {
            "types": "./__dist/mod.d.ts",
            "default": "./__dist/mod.js"
        },
        "./auth": {
            "types": "./__dist/auth/mod.d.ts",
            "default": "./__dist/auth/mod.js"
        },
        "./cache": {
            "types": "./__dist/cache/mod.d.ts",
            "default": "./__dist/cache/mod.js"
        },
        "./csp": {
            "types": "./__dist/csp/mod.d.ts",
            "default": "./__dist/csp/mod.js"
        },
        "./sec_fetch": {
            "types": "./__dist/sec_fetch/mod.d.ts",
            "default": "./__dist/sec_fetch/mod.js"
        }
    },
    "scripts": {
        "clean": "npm run clean:dist && npm run clean:tsbuildinfo",
        "clean:dist": "rm -rf __dist/",
        "clean:tsbuildinfo": "rm -rf tsconfig.tsbuildinfo",
        "prebuild": "npm run clean",
        "build": "tsc --project ."
    },
    "repository": {
        "type": "git",
        "url": "git+https://github.com/Nikkei/node-http-helper.git"
    },
    "author": "Nikkei Inc.",
    "license": "MIT",
    "bugs": {
        "url": "https://github.com/Nikkei/node-http-helper/issues"
    },
    "homepage": "https://github.com/Nikkei/node-http-helper",
    "dependencies": {
        "option-t": "^39.0.2"
    }
}
