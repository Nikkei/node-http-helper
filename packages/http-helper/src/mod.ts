export * as Auth from './auth/mod.js';
export * as Cache from './cache/mod.js';
export * as ClearSiteData from './clear_site_data.js';
export * as CSP from './csp/mod.js';
export * as HttpHeaderName from './http_header_name.js';
export * as HttpMethod from './http_method.js';
export * as HttpStatus from './http_status_code.js';
export * as Mime from './mime.js';
export { SecFetchDest, SecFetchMode, SecFetchSite, SecFetchUser } from './sec_fetch/mod.js';
