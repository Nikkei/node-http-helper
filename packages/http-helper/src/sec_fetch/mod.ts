// This implements https://www.w3.org/TR/fetch-metadata/

export * as SecFetchMode from './sec_fetch_mode.js';
