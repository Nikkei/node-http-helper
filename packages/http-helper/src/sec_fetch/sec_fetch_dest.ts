//  @see
//  - https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Sec-Fetch-Dest
//  - https://w3c.github.io/webappsec-fetch-metadata/#sec-fetch-dest-header
//  - https://fetch.spec.whatwg.org/#concept-request-destination
//  - https://fetch.spec.whatwg.org/#destination-table

export const EMPTY_STRING = '';

export const AUDIO = 'audio' as const;
export const AUDIO_WORKLET = 'audioworklet' as const;
export const DOCUMENT = 'document' as const;
export const EMBED = 'embed' as const;
export const FONT = 'font' as const;
export const FRAME = 'frame' as const;
export const IFRAME = 'iframe' as const;
export const IMAGE = 'image' as const;
export const MANIFEST = 'manifest' as const;
export const OBJECT = 'object' as const;
export const PAINT_WORKLET = 'paintworklet' as const;
export const REPORT = 'report' as const;
export const SCRIPT = 'script' as const;
export const SERVICE_WORKER = 'serviceworker' as const;
export const SHARED_WORKER = 'sharedworker' as const;
export const STYLE = 'style' as const;
export const TRACK = 'track' as const;
export const VIDEO = 'video' as const;
export const WEB_IDENTITY = 'webidentity' as const;
export const WORKER = 'worker' as const;
export const XSLT = 'xslt' as const;
