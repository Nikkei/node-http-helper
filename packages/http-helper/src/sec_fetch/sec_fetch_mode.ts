//  @see
//  - https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Sec-Fetch-Mode
//  - https://w3c.github.io/webappsec-fetch-metadata/#sec-fetch-mode-header
//  - https://fetch.spec.whatwg.org/#concept-request-mode

export const CORS = 'cors' as const;
export const NAVIGATE = 'navigate' as const;
export const NO_CORS = 'no-cors' as const;
export const SAME_ORIGIN = 'same-origin' as const;
export const WEBSOCKET = 'websocket' as const;
