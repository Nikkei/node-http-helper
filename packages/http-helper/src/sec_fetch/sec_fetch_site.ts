//  @see
//  - https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Sec-Fetch-Site
//  - https://w3c.github.io/webappsec-fetch-metadata/#sec-fetch-site-header

export const CROSS_SITE = 'cross-site' as const;
export const SAME_ORIGIN = 'same-origin' as const;
export const SAME_SITE = 'same-site' as const;
export const NONE = 'none' as const;
