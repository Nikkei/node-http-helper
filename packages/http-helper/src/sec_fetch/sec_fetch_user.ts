// @see
// - https://w3c.github.io/webappsec-fetch-metadata/#sec-fetch-user-header
// - https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Sec-Fetch-User
// - https://www.rfc-editor.org/rfc/rfc8941.html

import { SF_BOOLEAN_TRUE } from '../internals/structured_field_value/boolean.js';

/**
 *  By the spec, this value should be boolean `true` of structured field values format.
 *  - https://w3c.github.io/webappsec-fetch-metadata/#sec-fetch-user-header
 *  - https://www.rfc-editor.org/rfc/rfc8941.html#name-booleans
 */
export const TRUE = SF_BOOLEAN_TRUE;
