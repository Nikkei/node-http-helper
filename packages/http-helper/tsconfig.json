{
    "extends": "./tsconfig.base.json",
    "compilerOptions": {
        "outDir": "./__obj"
    },
    "include": [
        // @prettier-ignore
        "src/**/*",
        "__tests__/**/*"
    ]
}
