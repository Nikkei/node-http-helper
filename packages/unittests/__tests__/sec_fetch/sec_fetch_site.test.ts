import { test, expect } from 'vitest';

import { SecFetchSite } from '@nikkei/http-helper/sec_fetch';

test('SecFetchSite exported items', () => {
    expect(SecFetchSite).toMatchSnapshot();
});
