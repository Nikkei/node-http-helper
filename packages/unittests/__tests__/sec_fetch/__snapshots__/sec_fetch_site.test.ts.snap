// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SecFetchSite exported items 1`] = `
{
  "CROSS_SITE": "cross-site",
  "NONE": "none",
  "SAME_ORIGIN": "same-origin",
  "SAME_SITE": "same-site",
}
`;
